src/Common/MemoryTracker.cpp

```diff
@@ -442,6 +456,30 @@ bool MemoryTracker::updatePeak(Int64 will_be, bool log_memory_usage)
             && will_be / log_peak_memory_usage_every > peak_old / log_peak_memory_usage_every)
             logMemoryUsage(will_be);
 
+#if USE_JEMALLOC
+        if (level == VariableContext::Global && jemalloc_flush_profile_interval_bytes && DB::getJemallocValue<bool>("prof.active"))
+        {
+            static std::atomic<uint64_t> previous_flushed_peak = 0;
+            auto * flush_prefix = DB::getJemallocValue<char *>("opt.prof_prefix");
+            if (!flush_prefix)
+                LOG_WARNING(getLogger("MemoryTracker"), "Cannot flush memory profile, empty prefix");
```

Comment:
This can be done only if `log_memory_usage==true`, otherwise recursion is possible